lib/kubernetes-deploy/restart_task.rb

```diff
@@ -37,11 +38,14 @@ def perform!(deployments_names = nil, selector: nil)
       @logger.reset
 
       @logger.phase_heading("Initializing restart")
-      verify_namespace
-      deployments = identify_target_deployments(deployments_names, selector: selector)
-      if kubectl.server_version < Gem::Version.new(MIN_KUBE_VERSION)
-        @logger.warn(KubernetesDeploy::Errors.server_version_warning(kubectl.server_version))
+      errors = Validator.new(@task_config).errors
```

Comment:
Nit: can we make this name more specific? `TaskConfigValidator` by the looks of it? If this only validates the TaskConfig object though, why not make the interface `task_config.valid?` directly?